src/Symfony/Component/Encryption/Phpseclib/PhpseclibEncryption.php

```diff
@@ -0,0 +1,185 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Encryption\Phpseclib;
+
+use phpseclib\Crypt\AES;
+use phpseclib\Crypt\Random;
+use phpseclib\Crypt\RSA;
+use Symfony\Component\Encryption\Ciphertext;
+use Symfony\Component\Encryption\EncryptionInterface;
+use Symfony\Component\Encryption\Exception\DecryptionException;
+use Symfony\Component\Encryption\Exception\EncryptionException;
+use Symfony\Component\Encryption\Exception\InvalidKeyException;
+use Symfony\Component\Encryption\Exception\SignatureVerificationRequiredException;
+use Symfony\Component\Encryption\Exception\UnableToVerifySignatureException;
+use Symfony\Component\Encryption\Exception\UnsupportedAlgorithmException;
+use Symfony\Component\Encryption\KeyInterface;
+
+/**
+ * The secret key length should be 32 bytes, but other sizes are accepted.
+ *
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ *
+ * @experimental in 5.3
+ */
+class PhpseclibEncryption implements EncryptionInterface
+{
+    public function __construct()
+    {
+        if (!class_exists(RSA::class)) {
+            throw new \LogicException('You cannot use "Symfony\Component\Security\Core\Encryption\PhpseclibEncryption" as the "phpseclib/phpseclib:2.x" package is not installed. Try running "composer require phpseclib/phpseclib:^2".');
+        }
+    }
+
+    public function generateKey(string $secret = null): KeyInterface
+    {
+        $rsa = new RSA();
+        $key = $rsa->createKey();
+
+        if ($key['partialkey']) {
+            throw new EncryptionException('Failed to generate RSA keypair.');
+        }
+
+        if (null === $secret) {
+            $secret = random_bytes(32);
+        }
+
+        return PhpseclibKey::create($secret, $key['privatekey'], $key['publickey']);
+    }
+
+    public function encrypt(string $message, KeyInterface $myKey): string
+    {
+        if (!$myKey instanceof PhpseclibKey) {
+            throw new InvalidKeyException(sprintf('Class "%s" will only accept key objects of class "%s"', self::class, PhpseclibKey::class));
+        }
+
+        set_error_handler(__CLASS__.'::throwError');
+
+        try {
+            $aes = new AES(AES::MODE_CBC);
```

Comment:
Is it ok? I'm not expert in cryptography.
same for all other choices in this class. (iv length, RSA encryptionMode, SignatureMode)

Reply:
Sort answer: I think so, but Im not sure. 

This is basically copied from their documentation.